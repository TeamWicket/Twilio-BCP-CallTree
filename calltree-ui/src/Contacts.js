import React from 'react';
import { List, Edit, Datagrid, TextField, 
    SimpleForm, ReferenceField, required,
    TextInput, SelectInput, Create,
    regex, ReferenceInput } from 'react-admin';


const validateNumber = [required(),regex(/\+[0-9]/, 'Must be in format +9999999')];

export const FullNameField = ({ record = {} }) => <span>{record.firstName} {record.lastName}</span>;
FullNameField.defaultProps = { label: 'Name' };

export const ContactList = props => (
    <List {...props}>
        <Datagrid rowClick="edit">
            <TextField source="firstName" />
            <TextField source="lastName" />
            <TextField source="phoneNumber" />
            <TextField source="role" />
            <ReferenceField source="pointOfContactId" reference="contacts"><FullNameField source="firstName" /></ReferenceField>
        </Datagrid>
    </List>
);

export const ContactEdit = props => (
    <Edit {...props}>
        <SimpleForm>
            <TextInput source="firstName" />
            <TextInput source="lastName" />
            <TextInput source="phoneNumber" validate={validateNumber} />
            <SelectInput source="role" choices={[
                { id: 'CHAMPION', name: 'Champion' },
                { id: 'MANAGER', name: 'Manager' },
                { id: 'LEADER', name: 'Leader' },
                { id: 'REPORTER', name: 'Reporter'},
            ]}/>
            <ReferenceInput
                source="pointOfContactId"
                reference="contacts">
                <SelectInput optionText="firstName"/>
            </ReferenceInput>
        </SimpleForm>
    </Edit>
);

export const ContactCreate = props => (
    <Create {...props}>
        <SimpleForm redirect ="list">
            <TextInput source="firstName" validate={[required()]} />
            <TextInput source="lastName"  validate={[required()]} />
            <TextInput source="phoneNumber" validate={validateNumber} />
            <SelectInput source="role" choices={[
                { id: 'CHAMPION', name: 'Champion' },
                { id: 'MANAGER', name: 'Manager' },
                { id: 'LEADER', name: 'Leader' },
                { id: 'REPORTER', name: 'Reporter'},
            ]} validate={[required()]} />
            <ReferenceInput
                source="pointOfContactId"
                reference="contacts">
                <SelectInput optionText="firstName"/>
            </ReferenceInput>
        </SimpleForm>
    </Create>
);