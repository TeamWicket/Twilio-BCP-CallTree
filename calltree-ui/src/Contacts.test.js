import { ContactList, ContactEdit, ContactCreate } from './Contacts';
import React from 'react';
import { shallow } from 'enzyme';
describe("Contacts", () => {
    it("should match snapshot of ContactList", () => {
        expect(shallow(<ContactList />)).toMatchSnapshot();
    })
    it("should match snapshot of ContactEdit", () => {
        expect(shallow(<ContactEdit />)).toMatchSnapshot();
    })
    it("should match snapshot of ContactCreate", () => {
        expect(shallow(<ContactCreate />)).toMatchSnapshot();
    })
})
