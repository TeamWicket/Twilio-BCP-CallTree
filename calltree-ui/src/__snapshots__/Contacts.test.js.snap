// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts should match snapshot of ContactCreate 1`] = `ShallowWrapper {}`;

exports[`Contacts should match snapshot of ContactEdit 1`] = `ShallowWrapper {}`;

exports[`Contacts should match snapshot of ContactList 1`] = `ShallowWrapper {}`;
